### PR TITLE
add remap_default (issue 1157)

### PR DESCRIPTION
**Pull request by Tekkitslime**
This commit adds `remap_default`, which has an extra argument that is returned if `istart == istop`, that is, if there is a division by zero in `remap`, it returns the default argument.

## Summary by CodeRabbit

* **New Features**
  * Added remap_default() — safely remaps numeric values between ranges, returning a provided fallback when the result would be infinite or NaN.

* **Documentation**
  * Documented remap_default() with signature, behavior notes, and usage examples; updated remap entry to reference the safer variant.

* **Tests**
  * Added tests covering remap_default() across types, edge cases, infinities and NaN.

### CODE REVIEW

**Comment by JoltedJon** [comment]
@Tekkitslime Would you mind adding in some unit tests for this new function?

You can add the test to `tests/core/variant/test_variant_utility.h` 

You have to compile the project with unit tests enabled to test it by doing `scons tests=yes` and then running the project with `./bin/<redot_binary> --test`

File: core/variant/variant_utility.cpp [hunk 1]
```diff
@@ -556,6 +556,13 @@ double VariantUtilityFunctions::remap(double value, double istart, double istop,
 	return Math::remap(value, istart, istop, ostart, ostop);
 }
 
+double VariantUtilityFunctions::remap_default(double value, double istart, double istop, double ostart, double ostop, double default_value) {
+	if (istart == istop) {
```
Comment (Arctis-Fireblight) [inline core/variant/variant_utility.cpp]: Is this the only value combination that can produce NaN?
My thoughts would be to potentially call `Math::remap()` and then check the result for an errant result such as NaN or infinity, and return the default value under those conditions.

Comment (JoltedJon) [reply core/variant/variant_utility.cpp]: This also probably needs to use Math::is_approx equal due to floating point precision

**Comment by Tekkitslime** [comment]
There are more error conditions? like what? Maybe passing one of the arguments as NaN or Inf? Yeah i think you have a point, i should just check if remap gives a bad result and only then return the default value!

File: tests/core/variant/test_variant_utility.h [hunk 2]
```diff
@@ -32,6 +32,7 @@
 
 #pragma once
 
+#include "core/math/math_defs.h"
```
[the remark this answers is not in the document]
Comment (Arctis-Fireblight) [reply tests/core/variant/test_variant_utility.h:35]: Where are you getting this from Coderabbit?
The CI is building and passing all tests for all platforms?
